src/js/pages/catalog/PackageDetailTab.js

```diff
@@ -78,15 +99,31 @@ class PackageDetailTab extends mixin(StoreMixin) {
     });
   }
 
-  componentDidMount() {
-    super.componentDidMount(...arguments);
-
+  retrievePackageInfo() {
     const { packageName } = this.props.params;
     const { version } = this.props.location.query;
+
+    // fetch package versions available
+    CosmosPackagesStore.fetchPackageVersions(packageName);
     // Fetch package description
     CosmosPackagesStore.fetchPackageDescription(packageName, version);
   }
 
+  componentDidMount() {
+    super.componentDidMount(...arguments);
+    this.retrievePackageInfo();
+  }
+
+  componentDidUpdate(prevProps) {
```

Comment:
Why add this indirection between selecting a version on the dropdown and fetching new package info? I think it's more clear to just directly call `this.retrievePackageInfo()` after updating the query string.

Reply:
because this is part of the React lifecycle, no need to call directly since the mechanism is there already

Reply:
I know this works, I'm just saying that `componentDidUpdate` is being introduced only for this very specific url hash update...so I don't see why we would add more abstraction by using React's lifecycle for this.

Reply:
discussed offline